### PR TITLE
Fixed boolean and datetime filter options

### PR DESCRIPTION
**Pull request by habib535**
Some URLs to test.

http://localhost:8080/odataApi/api/ODataApi__Test__Sample?$filter=StartDate+gt+2006-12-30T23:59:59.99Z

http://localhost:8080/odataApi/api/ODataApi__Test__Sample?$filter=Active%20eq%20true

### CODE REVIEW

File: src/ODataApi/OData/Helpers/SqlBuilder.cs [hunk 1]
```diff
@@ -147,11 +147,16 @@ private SqlBuilder Compare(string @operator, string field, object value)
         {
             this.TryAddWhereClause();
             this.query.Append($" x.{field} {@operator} ?");
+            if (value.GetType().FullName.Contains("DateTimeOffset"))
```
Comment (un-tone) [inline src/ODataApi/OData/Helpers/SqlBuilder.cs]: Maybe something like this?

```cs
if (value is DateTimeOffset dataTimeOffset)
{
    value = dataTimeOffset.DateTime;
}
````

Comment (un-tone) [reply src/ODataApi/OData/Helpers/SqlBuilder.cs]: Also I guess `value` can be `null`, isn't it? If so, this `value.GetType()` can throw an exception.

**Comment by un-tone** [comment]
@habib535 I missed that the PR was targeted on `master-2.4` branch. Can you apply these changes for `master` branch also?

**Comment by habib535** [comment]
@un-tone thank you for noticing, the changes are merged.

**Pull request merged**